Assets/Scripts/Towers/Tower.cs

```diff
@@ -1,152 +1,207 @@
 using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.Events;
 
-public class Tower : MonoBehaviour {
-    public enum TargetingMethod {
-        HighestPriority,
-        LowestPriority,
-        Closest
-    }
-
-    public TargetingMethod targetingMethod;
-
-    public float damagePerShot;
-    public float shotCooldown;
-    public float range;
-
-    private float currentShotCooldown = 0.0f;
-
-    private HashSet<Enemy> enemiesInRange;
-    private Enemy currentTarget = null;
+public class EnemyHitEvent : UnityEvent<GameObject> {}
 
-    void Start() {
-        enemiesInRange = new HashSet<Enemy>();
-        currentShotCooldown = shotCooldown;
+public class PositionHitEvent : UnityEvent<Vector3> {}
 
-        CapsuleCollider capsuleCollider = GetComponent<CapsuleCollider>();
-
-        capsuleCollider.radius = range;
-        capsuleCollider.height = range * 4.0f;
+public class Tower : MonoBehaviour {
+  public TowerType towerType;
+  
+  private float currentShotCooldown = 0.0f;
+  
+  private List<Collider> enemiesInRange;
+  private Collider currentTarget = null;
+  
+  private Vector3 launchOrigin;
+  
+  void Start() {
+    enemiesInRange = new List<Collider>();
+    currentShotCooldown = towerType.shotCooldown;
+    
+    CapsuleCollider capsuleCollider = GetComponent<CapsuleCollider>();
+    
+    capsuleCollider.radius = towerType.range;
+    capsuleCollider.height = towerType.range * 4.0f;
+    
+    launchOrigin = transform.position;
+    
+    foreach (Transform child in transform) {
```

Comment:
Not sure if we need this that generic or if we can just make it a Vector3 we set by hand for each tower type

Reply:
My interest was to have the launch origin be a movable object in the editor, but I can change it no problem